### PR TITLE
remove duplicated global config commented out props on upgrade

### PR DESCRIPTION
**Pull request by oshi97**
This comes from a QA item where, after upgrading the theme, the global
config had duplicated "commented out" props. For example a commented out apiKey
like `// apiKey: <REPLACE_ME>`

The way comment-json works is that each comment is associated with a
specific "prop" of the json (unless the json is empty or the comment
is outside of the main json body).

If we're not careful with how we add new config to the global_config,
the prop associated with the giant commented out config block will change.
Because we merge the new theme's global_config with the user's current
global_config on upgrade (with the user's having higher priority), this
results in said huge block being duplicated, because comment-json sees
it as a *different* comment than the preexisting giant one, since it is
now associated with a different prop.

Update comment-json to latest for small parsing improvements.
There are no breaking changes we're worried about.

J=SLAP-1514
TEST=manual, auto

unit tests
test upgrading yanswers, also with some minor tweaks to their config
to test things

### CODE REVIEW

File: tests/postupgrade/utils.js [hunk 1]
```diff
@@ -1,17 +1,93 @@
...
+      }
+    ]);
+  });
+
+  it('ignores block comments, and comments that dont "look like" config props', () => {
+    const commentJson = parse(`{
+      // ignore me
+      "experienceVersion": "<REPLACE ME>",
+      /* ignore me too */
+      "businessId": "<REPLACE ME>",
```
Comment (oshi97) [inline tests/postupgrade/utils.js:63]: will remove the trailing comma

File: postupgrade/utils.js [hunk 2]
```diff
@@ -5,15 +5,77 @@ const simpleGit = require('simple-git/promise')();
...
+exports.mergeGlobalConfigs = function (newJson, originalJson) {
+  const originalParsed = parse(originalJson);
+  const commentsFromOriginal = parseCommentedOutProps(originalParsed);
+  const newPruned = pruneDuplicatedComments(parse(newJson), commentsFromOriginal);
+  const merged = assign(newPruned, originalParsed);
   return stringify(merged, null, 2);
 }
 
+/**
+ * Parses "commented out" config values.
```
Comment (tmeyer2115) [inline postupgrade/utils.js:23]: Does this method get all of the "commented out" config values? If so, I think we should make that a bit clearer in either the function name or JSDocs.

File: postupgrade/utils.js [hunk 3]
```diff
@@ -5,15 +5,77 @@ const simpleGit = require('simple-git/promise')();
...
+
+/**
+ * Removes all comments in CommentJSONValue that have the same value and type as the given
+ * commentsToDedupe.
+ *
+ * @param {import('comment-json').CommentJSONValue} commentJsonValue
+ * @param {import('comment-json').CommentToken[]} commentsToDedupe
+ * @returns {import('comment-json').CommentJSONValue} the updated value
+ */
+function pruneDuplicatedComments(commentJsonValue, commentsToDedupe) {
```
Comment (tmeyer2115) [inline postupgrade/utils.js:47]: Small nit: Should we just call this `pruneComments(parsedJson, comments)`? That might be a bit simpler. If I understand correctly, the purpose of this method is to ensure the provided comments are removed from the `parsedJson`. This method isn't concerned with why those comments need to be removed (e.g. they are dupes), just that they are removed.

**Review by tmeyer2115 — COMMENTED** [review]
Is this something we should file a bug report on to `comment-json`? From my understanding, this seems like something they should handle. But, maybe there is some nuance to our use case that I'm missing.

**Comment by oshi97** [comment]
> Is this something we should file a bug report on to `comment-json`? From my understanding, this seems like something they should handle. But, maybe there is some nuance to our use case that I'm missing.

I think this is more our specific business logic than what comment-json is responsible for. We could maybe ask them it as a feature, but I wouldn't consider this a bug in their code. At any rate they definitely won't be quick to add support, because it would be a breaking API change.

 In order to differentiate between
``` jsonc
{
  // Put your api key below
  "apiKey": "blah"
}
```

```jsonc
{
  // "anotherProp": "this is a commented out value, not a descriptor for apiKey",
  "apiKey": "blah"
}
```

They would need to assume that the second comment "looks like" a json value, rather than a comment describing apiKey. We can safely do this, because we know we want all comments that "look like json" to be unique. On the other hand I think comment-json was more intended for "comments describing json".

File: postupgrade/utils.js (hunk 3 above)
Comment (oshi97) [reply postupgrade/utils.js:47]: yup! I'll remove all instances of words like dedupe from this method

**Review by tmeyer2115 — APPROVED** [review]

**Pull request merged**